queue.go

```diff
@@ -178,14 +191,14 @@ func (q *DQue) Close() error {
 
 // Enqueue adds an item to the end of the queue
 func (q *DQue) Enqueue(obj interface{}) error {
-	if q.fileLock == nil {
-		return ErrQueueClosed
-	}
-
 	// This is heavy-handed but its safe
 	q.mutex.Lock()
 	defer q.mutex.Unlock()
 
+	if q.fileLock == nil {
+		return ErrQueueClosed
+	}
+
```

Comment:
I had to swap these two instruction blocks:
Old:
* check if queue is closed
* potentially somebody else calls close
* get the mutex lock
* access potentially closed queue

New:
* get the mutex lock
* nobody else can close this queue while we have the lock
* check if closed
* access a 100% open queue